components/datadog/apps/nginx/ecsFargate.go

```diff
@@ -0,0 +1,159 @@
+		Memory:    pulumi.IntPtr(96),
+		Essential: pulumi.BoolPtr(true),
+		DependsOn: ecs.TaskDefinitionContainerDependencyArray{
+			ecs.TaskDefinitionContainerDependencyArgs{
+				ContainerName: pulumi.String("datadog-agent"),
+				Condition:     pulumi.String("HEALTHY"),
+			},
+		},
+		PortMappings: ecs.TaskDefinitionPortMappingArray{
+			ecs.TaskDefinitionPortMappingArgs{
+				ContainerPort: pulumi.IntPtr(80),
+				HostPort:      pulumi.IntPtr(80),
+				Protocol:      pulumi.StringPtr("tcp"),
+			},
+		},
+		HealthCheck: ecs.TaskDefinitionHealthCheckArgs{
+			Command: pulumi.StringArray{
+				pulumi.String("CMD-SHELL"),
+				pulumi.String("apk add curl && curl --fail http://localhost || exit 1"),
+			},
+		},
+		LogConfiguration: ecsClient.GetFirelensLogConfiguration(pulumi.String("nginx"), pulumi.String("nginx"), apiKeySSMParamName),
+	}
+
+	serverTaskDef, err := ecsClient.FargateTaskDefinitionWithAgent(e, e.CommonNamer.ResourceName("nginx-server"), e.CommonNamer.DisplayName(255, pulumi.String("server")), 1024, 2048, map[string]ecs.TaskDefinitionContainerDefinitionArgs{"server": *serverContainer}, apiKeySSMParamName, fakeintake, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := ecs.NewFargateService(e.Ctx, namer.ResourceName("server"), &ecs.FargateServiceArgs{
+		Cluster:      clusterArn,
+		Name:         namer.DisplayName(255, pulumi.String("server")),
+		DesiredCount: pulumi.IntPtr(1),
+		NetworkConfiguration: classicECS.ServiceNetworkConfigurationArgs{
+			AssignPublicIp: pulumi.BoolPtr(e.ECSServicePublicIP()),
+			SecurityGroups: pulumi.ToStringArray(e.DefaultSecurityGroups()),
+			Subnets:        alb.LoadBalancer.Subnets(),
+		},
+		TaskDefinition:            serverTaskDef.TaskDefinition.Arn(),
+		EnableExecuteCommand:      pulumi.BoolPtr(true),
+		ContinueBeforeSteadyState: pulumi.BoolPtr(true),
+		LoadBalancers: classicECS.ServiceLoadBalancerArray{
+			&classicECS.ServiceLoadBalancerArgs{
+				ContainerName:  pulumi.String("server"),
+				ContainerPort:  pulumi.Int(80),
+				TargetGroupArn: alb.DefaultTargetGroup.Arn(),
+			},
+		},
+	}, opts...); err != nil {
+		return nil, err
+	}
+
+	queryContainer := &ecs.TaskDefinitionContainerDefinitionArgs{
+		Name:  e.CommonNamer.DisplayName(255, pulumi.String("query")),
+		Image: pulumi.String("ghcr.io/datadog/apps-http-client:main"),
+		Command: pulumi.StringArray{
+			pulumi.String("-url"),
+			pulumi.Sprintf("http://%s", alb.LoadBalancer.DnsName()),
+		},
+		Cpu:       pulumi.IntPtr(50),
+		Memory:    pulumi.IntPtr(32),
+		Essential: pulumi.BoolPtr(true),
+		DockerLabels: pulumi.StringMap{
+			"com.datadoghq.ad.tags": pulumi.String("[\"ecs_task_type:fargate\"]"),
+		},
+	}
+
+	queryTaskDef, err := ecsClient.FargateTaskDefinitionWithAgent(e, e.CommonNamer.ResourceName("nginx-query"), e.CommonNamer.DisplayName(255, pulumi.String("query")), 1024, 2048, map[string]ecs.TaskDefinitionContainerDefinitionArgs{"query": *queryContainer}, apiKeySSMParamName, fakeintake, opts...)
```

Comment:
It’s not needed to use a `namer` for the `name` and `family` parameters of `ecsClient.FargateTaskDefinitionWithAgent` because this function already uses a namer:
https://github.com/DataDog/test-infra-definitions/blob/40f6fda565a098ae6e4111bd95db84fd22952aac/resources/aws/ecs/fargateService.go#L42
https://github.com/DataDog/test-infra-definitions/blob/40f6fda565a098ae6e4111bd95db84fd22952aac/resources/aws/ecs/fargateService.go#L52